### ISSUE TITLE
normalization: scran vs neg. binomial regression (sctransform)

### ISSUE THREAD

**Issue by grst** [theislab/single-cell-tutorial#55]
Hi Malte, 

in the single-cell tutorial, you recommend scran as the go-to solution for normalization; however, in the recent single-cell webinar, you said you found using scran + NB regression worked best. As far as I understand [sctransform](https://github.com/ChristophH/sctransform) is such a NB regression method that is getting popular and [being added to scanpy](https://github.com/theislab/scanpy/pull/1271). 

![image](https://user-images.githubusercontent.com/7051479/102201079-eb2ce700-3ec5-11eb-8426-69027c66366d.png)

As of now, would you recommend to use sctransform rather than scran, or even both? If you use both, do you first adjust the counts with scran size factors and then feed them in the regression model? 

Best, 
Gregor

---

To get an idea, I applied CPM-normalization (top), scran, sctransform, scran+sctransform (bottom) on the Lung dataset from [Laughney et al](https://doi.org/10.1038/s41591-019-0750-6). All with default parameters. Just judging from these plots I don't like the result of sctransform too much: The CD8 and CD4 T cells are not clearly separated after this normalization any more. 
![image](https://user-images.githubusercontent.com/7051479/102227796-45d83a00-3eea-11eb-8e52-d15c247e2d43.png)

**Comment by LuckyMD** [comment]
Hey @grst,

This is a very good question, that I think can't really be answered without a good comparison. A week or so after I finished the last revision of the best practices paper, sctransform and GLM-PCA came out on preprint. So I couldn't even mention them. I quite like both methods (note the recent pearson residuals from the Kobak lab which argues sctransform overfits though), but there is currently no independent method comparison for these approaches to be able to say one is better... that's why it's hard to recommend one over another.

The case study where you show the picture from is on quite low quality data. It helped a lot there to separate out the cell states. I find it hard to assess whether sctransform is performing well or not in this data based on 3 marker genes... especially as there are also distinct batches, which you could incorporate into the sctransform model to maybe make it behave a bit better.

**Comment by grst** [comment]
Hi Malte, 

thanks for your swift reply! I was afraid that there's no definite way to tell... hopefully someone picks it up in a benchmark at some point :) 

Best, 
Gregor

**Comment by LuckyMD** [comment]
Agreed... would love a massive normalization benchmark... maybe that's a cool task to add to (CZI open problems)[https://github.com/singlecellopenproblems/SingleCellOpenProblems]

**Comment by vitkl** [comment]
Interesting discussion. I think the default parameters need to be modified depending on the tissue. For example, scTransform in Seurat does HVG selection for just 3000 genes which can be too low for many datasets like the one @grst is considering.

**Comment by LuckyMD** [comment]
I would love to know how you determine how many HVGs are good. For the best practices paper, I did a small survey of how many HVGs are used, and it was generally between 1000-5000, with most cases using fewer than 2500 if I recall correctly. I normally err on the side that more HVGs won't really harm you much (assuming < 10k or so I guess). But I don't have a way to say if a particular number is too low or too high.

**Comment by grst** [comment]
In fact, I used 4000 HVGs for the above analysis (see the [corresponding gist](https://gist.github.com/grst/6c51648cccd10d57eb9676f5e75f2216). 

I tend to choose more HVGs the more cell-types a dataset contains (i.e. 4k-6k for whole tissue; for datasets with only T cells I went down to 2k because I found that it reduces batch effects). But this is all based on gut feeling and I would also be interested in an empirical way to determine a reasonable HVG cutoff.
